# Enteros (INT) 
Cree una base de datos ` prueba` y dentro de ella una tabla ` T1` con una columna ` numero` de tipo entero

In [8]:
CREATE DATABASE PRUEBA

GO

--??? seleccionar la base de datos recién creada
USE PRUEBA
--Crear la tabla especificada
CREATE TABLE T1 (
    numero int
)


Commands completed successfully.

Commands completed successfully.

Total execution time: 00:00:01.941

In [9]:
select * from T1

(0 rows affected)

Total execution time: 00:00:00.021

numero

Inserte filas en la tabla ` T1` con los valores enteros 100, 200, 150

In [10]:
--insert into ????
INSERT INTO  T1 VALUES (100),(200),(150) 
select * from T1


(3 rows affected)

(3 rows affected)

Total execution time: 00:00:00.773

numero 100 200 150

Ahora inserte una fila con los valores reales 99.99, 99.05

In [11]:
insert into T1  VALUES(99.99),(99.05)

select * from T1

(2 rows affected)

(5 rows affected)

Total execution time: 00:00:00.083

numero 100 200 150 99 99

**¿Que sucedió?: Se tomó solamente el valor entero de el valor en cuestión, ignorando los valores decimales (fueron trunqueados).

# Cadenas (VARCHAR)

Cree ahora una tabla `T2` con una columna `cadena` de tipo `CHAR(10)` y otra `cadenavar` de tipo `VARCHAR(10)`. Luego inserte los valores 'Corto', 'Cadena larga' para ambas columnas.

In [5]:
--CREATE ?? ??
CREATE TABLE T2 (
   cadena char(10),
   cadenavar varchar(10)
)

INSERT into T2 values ('Corto', 'Corto'), ('Cadenota', 'Cadenota' )


(2 rows affected)

Total execution time: 00:00:00.149

**¿Que sucedió?**


Corrija e intente de nuevo

Nota: La longitud de 'Cadena Larga' es mayor al límite establecido al cadenavar. Se cambió la cadena por una adecuada.

¿Cuanto espacio de almacenamiento usa cada dato?



In [6]:
SELECT len(cadena) as 'Longitud de cadena', DATALENGTH(cadena) as 'Espacio usado por cadena', 
       len(cadenavar) as 'Longitud de cadenavar', DATALENGTH(cadenavar) as 'Espacio usado por cadenavar'
    from T2

(2 rows affected)

Total execution time: 00:00:00.123

Longitud de cadena Espacio usado por cadena Longitud de cadenavar Espacio usado por cadenavar 5 10 5 5 8 10 8 8

# REAL

Usemos ahora el tipo de dato `REAL`. Cree una tabla `T3` con una columna `precio` de tipo `REAL`. Agregue 10 filas con el valor 0.1

In [13]:
-- Crear la tabla
CREATE TABLE T3 (
    precio REAL
)
INSERT INTO T3 VALUES (0.1),(0.1),(0.1),(0.1),(0.1),(0.1),(0.1),(0.1),(0.1),(0.1)
select sum(precio) from T3


(10 rows affected)

(1 row affected)

Total execution time: 00:00:00.070

(No column name) 1.00000001490116

**¿Que resultado se obtiene si sumamos los valores de la columna precio?:**
1.00000001490116

***
**Por que?:** 
El tipo de dato REAL maneja una base binaria, por lo cual no hay exactitud en cuanto a resultados enteros.
***

# Moneda (Numeric) #

Cree una tabla `T4` con una columna `precio` de tipo `numeric(10,4)`. Agregue 10 filas con el valor 0.1

In [2]:
-- Crear la tabla
CREATE TABLE T4(
    precio NUMERIC(10,4)
)
INSERT INTO T4 VALUES(0.1),(0.1),(0.1),(0.1),(0.1),(0.1),(0.1),(0.1),(0.1),(0.1)

select sum(precio) from T4


(10 rows affected)

(1 row affected)

Total execution time: 00:00:00.240

(No column name) 1.0000

**¿Que resultado se obtiene ahora si sumamos los valores de la columna precio?:**
1.0000

***
**Por que?:** 
Por el contrario del valor REAL, este no maneja una base binaria para ser utilizado, pero si es importante manejarlo especialmente, ya que se definen ciertos formatos. En este caso solo se aceptarán valores numericos que tengan máximo 10 carácteres que representen enteros y 4 carácteres que representen los decimales.
***